src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -1301,21 +1302,17 @@ private IEnumerable<MemberDeclarationSyntax> AsMembersOf(SyntaxNode declaration,
 
         private SyntaxNode AsMemberOf(SyntaxNode declaration, SyntaxNode member)
         {
-            switch (declaration.Kind())
+            switch (declaration)
             {
-                case SyntaxKind.ClassDeclaration:
-                    var cd = (ClassDeclarationSyntax)declaration;
-                    return this.AsClassMember(member, cd.Identifier.Text);
-                case SyntaxKind.StructDeclaration:
-                    var sd = (StructDeclarationSyntax)declaration;
-                    return this.AsClassMember(member, sd.Identifier.Text);
-                case SyntaxKind.InterfaceDeclaration:
+                case TypeDeclarationSyntax typeDeclaration:
+                    return this.AsClassMember(member, typeDeclaration.Identifier.Text);
+                case InterfaceDeclarationSyntax:
```

Comment:
!!This is a BUG I noticed now!!
Please don't merge. I'll update it after CI finishes to see if any unit tests catches it or not.

It would be good if there is an analyzer to point to this. This is a dead conditional code and something that shouldn't be too difficult by an analyzer.

@Evangelink @mavasani Is this something suitable to be handled by CAxxxx or IDExxxx?